src/EditorFeatures/Core/Implementation/Diagnostics/SuggestionTag.cs

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Windows.Media;
+using System.Windows.Shapes;
+using Microsoft.CodeAnalysis.Editor.Implementation.Adornments;
+using Microsoft.VisualStudio.Text.Editor;
+
+namespace Microsoft.CodeAnalysis.Editor.Implementation.Diagnostics
+{
+    /// <summary>
+    /// Tag that specifies line separator.
+    /// </summary>
+    internal class SuggestionTag : GraphicsTag
+    {
+        public static readonly SuggestionTag Instance = new SuggestionTag();
+
+        private static Pen s_pen;
+
+        protected override void Initialize(IWpfTextView view)
+        {
+            base.Initialize(view);
+
+            if (s_pen != null)
+            {
+                return;
+            }
+
+            var color = Color.FromArgb(100, VerticalRuleColor.R, VerticalRuleColor.G, VerticalRuleColor.B);
+            s_pen = new Pen
+            {
+                Brush = new SolidColorBrush(color),
+                DashStyle = DashStyles.Dot,
+                DashCap = PenLineCap.Round,
+                Thickness = 2,
+            };
+        }
+
+        public override GraphicsResult GetGraphics(IWpfTextView view, Geometry geometry)
+        {
+            Initialize(view);
+
+            // We clip off a bit off the start of the line to prevent a half-square being
+            // drawn.
+            var clipRectangle = geometry.Bounds;
+            clipRectangle.Offset(1, 0);
```

Comment:
This doesn't seem like it should be necessary - can we get a WPF expert to weigh in here?